### PR TITLE
Write the doc page on profiling cpu events

### PR DESCRIPTION
**Pull request by bjacob**

### CODE REVIEW

File: docs/developing_iree/profiling_cpu_events.md [hunk 1]
```diff
@@ -1,13 +1,247 @@
...
+
+#### These events are hardware-dependent and under-documented
+
+Many of these events have a meaning that varies between CPUs and that is
+difficult to characterize on any CPU, let alone in a way that applies to all
+CPUs.
+
+For example, take the "L2 data cache refill". On ARM, with `simpleperf`, that
+would be `raw-l2d-cache-refill`. Questions:
+* Is “L2” inclusive of “L1”?
```
Comment (silvasean) [inline docs/developing_iree/profiling_cpu_events.md:201]: link to https://en.wikipedia.org/wiki/Cache_inclusion_policy

File: docs/developing_iree/profiling_cpu_events.md [hunk 2]
```diff
@@ -1,13 +1,247 @@
...
+which event, which will increase inaccuracy of per-symbol breakdowns for very
+cheap symbols (and makes `perf annotate` impossible as noted above). Finally, be
+aware that some ARM CPUs may perform event count interpolation, so we may not
+have any access to true hardware counts.
+
+### Recommendations
+
+Here is a workflow pattern that allows to make significant use of CPU event
+counts, despite all the problems noted above:
+* Hypothetize that some code diff might help performance, and might help
```
Comment (silvasean) [inline docs/developing_iree/profiling_cpu_events.md:222]: typo: hypothesize

File: docs/developing_iree/profiling_cpu_events.md [hunk 3]
```diff
@@ -1,13 +1,247 @@
...
+  reducing the number of CPU events of a certain type, and that the two might be
+  related.
+* Benchmark with and without the code diff, on the same device, everything else
+  being equal.
+  * Let your benchmark performa a fixed number of iterations, or, if using a
```
Comment (silvasean) [inline docs/developing_iree/profiling_cpu_events.md:227]: typo: performa

**Review by silvasean — APPROVED** [review]
Looks great. Thanks!

**Review by ScottTodd — APPROVED** [review]

File: docs/developing_iree/profiling_cpu_events.md (hunk 1 above)
Comment (bjacob) [reply docs/developing_iree/profiling_cpu_events.md:201]: done, thanks!

**Pull request merged**